### PR TITLE
Fix multitile engine rotating.

### LINKED ISSUE #51892: pubby monkey shuttle has problem when rotated
pubby monkey shuttle bug

### PR DESCRIPTION
**Pull request by Dennok**
## About The Pull Request

Multitile engines correctly rotates on shuttle rotating.
fix #51892
image
image

Now it must properly rotate any multitile objects, exept for BSA and Gatewey, it has non zero bound_x.

## Why It's Good For The Game

Bugs is bad

## Changelog
:cl:
fix: Now engines correctly rotate on shuttle rotating.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Perfection

File: code/modules/shuttle/shuttle_rotate.dm [hunk 1]
```diff
@@ -101,3 +101,22 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
 	. = ..()
 	if(wall_turret_direction && (params & ROTATE_DIR))
 		wall_turret_direction = turn(wall_turret_direction,rotation)
+
+/obj/structure/shuttle/shuttleRotate(rotation, params)
+	dir = angle2dir(rotation+dir2angle(dir))
```
Comment (ShizCalev) [inline code/modules/shuttle/shuttle_rotate.dm:106]: setDir

**Review by ShizCalev — DISMISSED** [review]

File: code/modules/shuttle/shuttle_rotate.dm (hunk 1 above)
Comment (Dennok) [reply code/modules/shuttle/shuttle_rotate.dm]: Done

**Comment by Dennok** [comment]
Now it must properly rotate any multitile objects, exept for BSA and Gatewey, it has non zero bound_x.

File: code/modules/shuttle/shuttle_rotate.dm [hunk 2]
```diff
@@ -25,6 +25,30 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
...
+			pixel_y = 0
+		if(dir == 8)
+			pixel_x = 0
+			pixel_y = -bound_height+world.icon_size
+		if(dir == 1)
+			pixel_x = -bound_width+world.icon_size
+			pixel_y = -bound_height+world.icon_size
+		if(dir == 4)
+			pixel_x = -bound_width+world.icon_size
+			pixel_y = 0
```
Comment (Rohesie) [inline code/modules/shuttle/shuttle_rotate.dm:48]:
```suggestion
		switch(dir)
			if(SOUTH)
				pixel_x = 0
				pixel_y = 0
			if(WEST)
				pixel_x = 0
				pixel_y = -bound_height+world.icon_size
			if(NORTH)
				pixel_x = -bound_width+world.icon_size
				pixel_y = -bound_height+world.icon_size
			if(EAST)
				pixel_x = -bound_width+world.icon_size
				pixel_y = 0
```

File: code/modules/shuttle/shuttle_rotate.dm [hunk 3]
```diff
@@ -25,6 +25,31 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
...
+				pixel_y = 0
+			if(WEST)
+				pixel_x = 0
+				pixel_y = -bound_height+world.icon_size
+			if(NORTH)
+				pixel_x = -bound_width+world.icon_size
+				pixel_y = -bound_height+world.icon_size
+			if(EAST)
+				pixel_x = -bound_width+world.icon_size
+				pixel_y = 0
```
Comment (Rohesie) [inline code/modules/shuttle/shuttle_rotate.dm:49]: There's also this possibility:
```suggestion
		pixel_x = dir & (NORTH|EAST) ? -bound_width+world.icon_size : 0
		pixel_y = dir & (NORTH|WEST) ? -bound_width+world.icon_size : 0
```

**Comment by Dennok** [comment]
Thanks, i thinked about this, but something gone wrong, and i drop it in this merely working state.

File: code/modules/shuttle/shuttle_rotate.dm (hunk 3 above)
Comment (Dennok) [reply code/modules/shuttle/shuttle_rotate.dm]: I like this more.

**Comment by Rohesie** [comment]
Is it tested and working?

**Comment by Dennok** [comment]
As i show above it work fine,
I hope I haven't missed anything
anyway nothing critical was touched

But i dont tested it after this optimization changes, but visually it must work right.
And i dont aclually tested pubby monkey shuttle, but the engine itself tested above.

**Pull request merged**